### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by dwram**
### Summary

Implemented Oauth feature allowing users to login with 3rd party applications like Facebook, Github and Twitter.

**Ancillary changes**

Updated Travis.yml to pass CI/CD.
Removed redundant duplicate SQL queries failing CI/CD
Github application environment variables are stored in Travis / Locally.

**Requires**

Bundle and possibly migration via db:migration

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+    if auth_hash
+      return flash[:notice] = 'Your email address must be publicly visible to use OAuth services' if auth_hash[:info][:email].nil?
+
+      #       User.find_or_create_by!(email: auth_hash[:info][:email], provider: auth_hash[:provider], uid: auth_hash[:uid]) do |user|
+      #         user.first_name = auth_hash[:info][:name]
+      #         user.password = 'mUc3m00R'
+      #         user.password_confirmation = 'mUc3m00R'
+      #         user.save
+      #         session[:id] = user.id
+      #       end
```
Comment (ralphm10) [inline app/controllers/sessions_controller.rb:17]: can commented out lines be removed?

Comment (02ship) [inline app/controllers/sessions_controller.rb:17]: We need to run the site from other machines (than Daniel's) before we can tell whether it is required or not, correct?

Comment (dwram) [reply app/controllers/sessions_controller.rb:17]: Unfortunately not. We may need to set a slug password after others have tested the login feature, requiring those lines to be uncommented.

Comment (dwram) [reply app/controllers/sessions_controller.rb:17]: Precisely.

File: config/application.rb [hunk 2]
```diff
@@ -10,7 +10,10 @@ module Acebook
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 5.1
-
+    config.middleware.use OmniAuth::Builder do
+      #provider :developer if Rails.env.development?
```
Comment (02ship) [inline config/application.rb:14]: Should this be removed entirely?

**Pull request merged**